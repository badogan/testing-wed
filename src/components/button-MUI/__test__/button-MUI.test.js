import React from 'react'
import ReactDOM from 'react-dom'
import ButtonMUI from '../button-MUI'
import {render, cleanup} from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'


afterEach(cleanup) //In fact not necessary

it("renders without crashing",()=>{
    const div = document.createElement("div")
    ReactDOM.render(<ButtonMUI></ButtonMUI>,div)
})

it("renders MUI-Button correctly", ()=>{
    // const {getByTestId} = render(<Button label="click me please"/>)
    const basri = render(<ButtonMUI/>)
    expect(basri.getByTestId("button-MUI")).toHaveTextContent("Hello World")
})

it("MUI-Button matches snapshot",()=>{
    const tree = renderer.create(<ButtonMUI/>).toJSON()
    expect(tree).toMatchSnapshot()
})