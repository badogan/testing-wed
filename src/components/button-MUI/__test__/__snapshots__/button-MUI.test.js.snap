// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MUI-Button matches snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
  data-testid="button-MUI"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Hello World
  </span>
</button>
`;
